pylint_django/checkers/models.py

```diff
@@ -107,7 +124,7 @@ def _visit_classdef(self, node):
 
         # a different warning is emitted if a parent declares __unicode__
         for method in node.methods():
-            if method.name == '__unicode__':
+            if method.parent != node and _is_unicode_or_str_in_python_2_compatibility(method):
```

Comment:
Isn't `method.parent != node` always True in this context ? This is what the comments suggest?

Also there's a striking similarity between the decorators loop used in your function and the decorators loop used just below this block of code (not shown here). 

Can you explain what you are trying to do because I'm not well familiar with this particular part of the code?

Reply:
> Isn't method.parent != node always True in this context ? This is what the comments suggest?

Not necessarily. The `methods()` generator contains both own and inherited methods. It works for the previous use case because it already asserted that there was no `__unicode__` method in the current class, so if it found one it had to be in one of the ancestors.

> Also there's a striking similarity between the decorators loop used in your function and the decorators loop used just below this block of code (not shown here).

Yes, they are handling two issues derived from the use of the same decorator. I should extract that iteration out. The original code checks for the existence of a decorator defined in the current class. If the decorator exists, it avoids emitting `model-missing-unicode`.

This fix handles the case where the `__unicode__` method is defined not explicitly in an ancestor class but using the decorator. Previously, it wrongly emitted `model-missing-unicode`. Now it emits `model-no-explicit-unicode` because the `__unicode__` method is defined by the decorator in the ancestor class.